File: app/src/main/java/com/krystianwsul/checkme/domainmodel/ScheduleBridge.java
package com.krystianwsul.checkme.domainmodel;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.util.Pair;

import com.krystianwsul.checkme.utils.TaskKey;

public interface ScheduleBridge {
    long getStartTime();

    @Nullable
    Long getEndTime();

    void setEndTime(long endTime);

    @NonNull
    TaskKey getRootTaskKey();

    void delete();

    @Nullable
    Pair<String, String> getRemoteCustomTimeKey();
}

File: app/src/main/java/com/krystianwsul/checkme/domainmodel/ScheduleBridge.kt
package com.krystianwsul.checkme.domainmodel

import com.krystianwsul.checkme.utils.CustomTimeKey
import com.krystianwsul.checkme.utils.RemoteCustomTimeId
import com.krystianwsul.checkme.utils.ScheduleId
import com.krystianwsul.checkme.utils.TaskKey

interface ScheduleBridge {
    
    val startTime: Long

    var endTime: Long?

    val rootTaskKey: TaskKey

    val remoteCustomTimeKey: Pair<String, RemoteCustomTimeId>?

    fun delete()

    val customTimeKey: CustomTimeKey<*>?

    val hour: Int?

    val minute: Int?

    val scheduleId: ScheduleId
}
